File: peerDiscoveryServer/src/main/java/it/uniroma3/cheatchat/peerdiscoveryserver/domain/Peer.java
package it.uniroma3.cheatchat.peerdiscoveryserver.domain;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@Entity
public class Peer {

    @Id
    private String username;
    private String ip;
    private LocalDateTime lastSeen;

    public int hashCode() {
        return this.username.hashCode();
    }

    public boolean equals(Object obj) {
        if (obj == null) {
            return false;
        }
        if (obj instanceof Peer) {
            Peer peer = (Peer) obj;
            return this.username.equals(peer.username);
        }
        return false;
    }
}

File: ccd/peer.py
class Peer:

    username : str
    ip : str
    last_seen : float

    def __init__ (self, username:str, ip:str, last_seen:float):
        self.username = username
        self.ip = ip
        self.last_seen = last_seen
    
    def to_string(self) -> str:
        return self.username + " " + self.ip + " " + str(self.last_seen)

    def __eq__(self, peer):
        return self.username == peer.username and self.ip == peer.ip

